### PR TITLE
Add pre-commit hook for validating release notes

### PR DESCRIPTION
**Pull request by aaradhak**
This is a pre-commit hook using `pre-commit` framework to validate release notes YAML files. 
- The hook checks for correct structure, formatting and required fields in the `release-notes` directory. 
- This ensures consistency in release notes before committing changes.

_Ref:_ 
- This pre commit check was pending as part of this [PR-934](https://github.com/coreos/fedora-coreos-streams/pull/934#issuecomment-2260059748)
- https://github.com/coreos/fedora-coreos-streams/issues/628

### CODE REVIEW

**Comment by aaradhak** [comment]
This pre commit work was pending as part of this [PR-934](https://github.com/coreos/fedora-coreos-streams/pull/934).

**Review by jlebon — COMMENTED** [review]
Not against this, but personally I don't feel like installing pre-commit just for this repo.

How about we also add it to the existing `make syntax-check` target?

**Comment by aaradhak** [comment]
ok, I have added it to the make syntax-check target.

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,8 @@
-.PHONY: syntax-check
+.PHONY: syntax-check pre-commit-check print-rollouts
```
Comment (travier) [inline Makefile:1]: Hum, why is this adding `pre-commit-check` here?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+repos:
```
Comment (travier) [inline .pre-commit-config.yaml:1]: What's using this file?

Comment (aaradhak) [reply .pre-commit-config.yaml:1]: This yaml is used by the pre-commit tool framework for managing pre-commit hooks. Pre-commit hooks run automatically before a commit is finalized. It requires the installation of pre-commit tool in the local environment

File: Makefile (hunk 1 above)
Comment (aaradhak) [reply Makefile:1]: based on the review comment, I added the pre-commit check to the syntax check here.

File: Makefile [hunk 3]
```diff
@@ -1,7 +1,8 @@
-.PHONY: syntax-check
-syntax-check:
-	@find streams updates -iname '*.json' | sort | xargs -n 1 python3 -c 'import json, sys; json.load(open(sys.argv[1]))'
+.PHONY: syntax-check pre-commit-check print-rollouts
 
-.PHONY: print-rollouts
+syntax-check: pre-commit-check
+	@find streams updates -iname '*.json' | sort | xargs -n 1 python3 -c 'import json, sys; json.load(open(sys.argv[1]))'
+pre-commit-check:
+	pre-commit run --all-files
```
Comment (jlebon) [inline Makefile:6]:
```suggestion
syntax-check: pre-commit-check
	@find streams updates -iname '*.json' | sort | xargs -n 1 python3 -c 'import json, sys; json.load(open(sys.argv[1]))'
	@find release-notes -iname '*.yaml' | sort | xargs -n1 python3 ci/check-release-notes.py
```

Comment (aaradhak) [reply Makefile]: thanks for this input. I have made these changes.

File: Makefile [hunk 4]
```diff
@@ -1,7 +1,7 @@
-.PHONY: syntax-check
-syntax-check:
-	@find streams updates -iname '*.json' | sort | xargs -n 1 python3 -c 'import json, sys; json.load(open(sys.argv[1]))'
+.PHONY: syntax-check pre-commit-check print-rollouts
```
Comment (jlebon) [inline Makefile:1]: Drop `pre-commit-check`.

**Review by jlebon — COMMENTED** [review]
Since we now have it part of `syntax-check`, it's already covered by the existing `validate.yml` workflow, so we can get rid of the `validate-release-notes.yml` file.

Coincidentally, it looks like validation is failing. :)

File: Makefile [hunk 5]
```diff
@@ -1,7 +1,7 @@
-.PHONY: syntax-check
+.PHONY: syntax-check print-rollouts
+
 syntax-check:
 	@find streams updates -iname '*.json' | sort | xargs -n 1 python3 -c 'import json, sys; json.load(open(sys.argv[1]))'
-
-.PHONY: print-rollouts
+	@find release-notes -iname '*.yaml' | sort | xargs -n1 python3 ci/check-release-notes.py
 print-rollouts:
```
[the remark this answers is not in the document]
Comment (aaradhak) [reply Makefile:7]: I have added a line above the print-rollouts

**Review by jlebon — APPROVED** [review]
Thanks! LGTM though let's squash to a single commit?

**Review by travier — APPROVED** [review]

**Pull request merged**